src/me/mrCookieSlime/Slimefun/api/TickerTask.java

```diff
@@ -202,20 +206,29 @@ public void info(CommandSender sender) {
 			tellraw.addText("   &7&oHover for more Info");
 			StringBuilder hover = new StringBuilder();
 			int hidden = 0;
-			
-			for (Map.Entry<String, Integer> entry: machineCount.entrySet()) {
-				long timings = machineTimings.get(entry.getKey());
-				if (timings > 0) hover.append("\n&c").append(entry.getKey()).append(" - ").append(entry.getValue()).append("x &7(").append(timings).append("ms)");
-				else hidden++;
+
+			Map<String, Long> timings = machineCount.keySet().stream()
+					.map(key -> new AbstractMap.SimpleEntry<>(key, machineTimings.getOrDefault(key, 0L)))
+					.sorted((o1, o2) -> o2.getValue().compareTo(o1.getValue()))
```

Comment:
Couldn't you just use Map.Entry.comparingByValue here too?

Reply:
Yeah I had it as that but good ol' Java generics got quite messed up and (despite being fine) failed compilation.